feature/main/src/main/java/com/ku_stacks/ku_ring/main/search/compose/SearchScreen.kt

```diff
@@ -89,10 +96,12 @@ private fun SearchScreen(
     onNavigationClick: () -> Unit,
     onSearch: (SearchState) -> Unit,
     onClickNotice: (Notice) -> Unit,
+    onClickClearSearchHistory: () -> Unit,
     searchState: SearchState,
     tabPages: List<SearchTabInfo>,
     noticeList: List<Notice>,
     staffList: List<Staff>,
+    keywordHistories: List<String>,
```

Comment:
Immutable Collections 사용해서 불필요한 리컴포지션 방지할 수 있을 것 같습니다.

```suggestion
    keywordHistories: ImmutableList<String>,
```

Reply:
검색어가 한 글자 입력될 때마다 recomposition이 발생하긴 하네요. 

그런데 현재 검색 화면에 성능 문제가 있지는 않은 것 같아서, 나중에 kotlinx immutable collections 도입할 때 한번에 작업해도 되지 않을까 싶습니다. 

참고로 현재 프로젝트에서 사용할 수 있는 `ImmutableList`는 firebase-crashlytics 라이브러리에 선언되어 있는 모조품(??)입니다. 진품은 [여기](https://github.com/Kotlin/kotlinx.collections.immutable) 있습니다.

Reply:
불필요한 리컴포지션이라는 것을 잘 이해 못했어요. keywordHistories 가 바뀔땐 필요한 recomposition 아닐까요?

Reply:
> 검색어 한 글자 입력 → 키워드 히스토리는 바뀌지 않았지만 recomposition됨

등이 있습니다.

Reply:
keywordHistory가 바뀌지 않았지만 `KeywordHistorySection`이 recompose되는 몇 가지 경우가 있는데요

1. 검색어를 한 글자 입력
2. 검색어가 바뀌었으므로 `SerachTextField`는 당연히 recompose됨
3. 검색어 외의 데이터는 바뀌지 않았으므로, 나머지 `KeywordHistorySection` 등은 recompose되지 않아야 함
4. 그러나 `List`는 unstable한 클래스 (compose 컴파일러가 데이터의 동일성을 의심함)
5. 따라서 `SearchScreen`의 자식 composable 중 `List`를 매개변수로 받는 것들도 (불필요하게) recompose됨
6. `KeywordHistorySection`도 `List` 타입 매개변수를 받으므로 recompose됨

`ImmutableList`는 compose에서 stable한 타입으로 인식하기 때문에, `List` 대신 `ImmutableList`를 사용하면 불필요한 recompososition을 줄일 수 있습니다.

현우님의 첫 코멘트는 이 문제를 고치자는 의견이었고, 저는 (문제 인식은 하되) 지금 이 PR에서 고칠 필요는 없다고 말씀드린 거였어요

Reply:
list 가 실제론 immutable 하더라도 compose 단에서는 stable 한지알수 없어서 불필요한 KeywordHistorySection까지 모두 recompose 된다는것이죠? 상세한 설명 감사합니다 👍

recomposition으로 눈에 띄게 성능이 체감되지는 않고, dependency 를 추가해야해서 이왕이면 별도 PR 로 다른 Feature 모듈까지 의존성추가하면 좋을것 같은데요,
별도 PR로 가져가는게 어떨까요?

Reply:
네, 저도 출시 후에 다시 보자는 의견입니다